src/librustc_mir/shim.rs

```diff
@@ -41,6 +41,15 @@ fn make_shim<'tcx>(tcx: TyCtxt<'tcx>, instance: ty::InstanceDef<'tcx>) -> &'tcx
                 None,
             )
         }
+        ty::InstanceDef::ReifyShim(def_id) => {
+            build_call_shim(
+                tcx,
+                def_id,
+                Adjustment::Identity, // TODO(anp) is this the right kind of adjustment?
```

Comment:
Yeah. You could even fold this with the `Virtual` case, both of them just need to generate a simple direct call.